recurring_contract/model/contract_group.py

```diff
@@ -34,89 +35,75 @@ def _get_change_methods(self, cr, uid, context=None):
              'Clean invoices')
         ]
 
-    def __get_change_methods(self, cr, uid, context=None):
+    def __get_change_methods(self):
         """ Call method which can be inherited """
-        return self._get_change_methods(cr, uid, context=context)
+        return self._get_change_methods()
 
     def _get_gen_states(self):
         return ['active']
 
-    def _get_next_invoice_date(self, cr, uid, ids, name, args, context=None):
-        res = {}
-        for group in self.browse(cr, uid, ids, context):
-            res[group.id] = min([c.next_invoice_date
-                                 for c in group.contract_ids
-                                 if c.state in self._get_gen_states()] or
-                                [False])
-        return res
+    @api.depends('contract_ids.next_invoice_date', 'contract_ids.state')
+    def _get_next_invoice_date(self):
+        next_inv_date = min(
+            [c.next_invoice_date for c in self.contract_ids
+             if c.state in self._get_gen_states()] or [False])
+        self.next_invoice_date = next_inv_date
 
-    def _get_last_paid_invoice(self, cr, uid, ids, name, args, context=None):
+    def _get_last_paid_invoice(self):
         res = dict()
-        for group in self.browse(cr, uid, ids, context):
+        for group in self:

```

Comment:
I think here instead of returning res you should assign the result in each iteration doing `group.last_paid_invoice_date = max(...)`